src/librustc/middle/expr_use_visitor.rs

```diff
@@ -1283,10 +1283,11 @@ impl<'d,'t,'tcx,TYPER:mc::Typer<'tcx>> ExprUseVisitor<'d,'t,'tcx,TYPER> {
     }
 }
 
-fn copy_or_move<'tcx>(typer: &mc::Typer<'tcx>,
-                      cmt: &mc::cmt<'tcx>,
-                      move_reason: MoveReason)
-                      -> ConsumeMode
+fn copy_or_move<'tcx,T>(typer: &T,
+                        cmt: &mc::cmt<'tcx>,
+                        move_reason: MoveReason)
+                        -> ConsumeMode
+    where T: mc::Typer<'tcx>
```

Comment:
Ah, yes, I remember the issue I had now.  Yeah, there was literally no way to add a closure (that I could see) without breaking object safety.